assets/js/payment-method-extensions/payment-methods/cheque/index.js

```diff
@@ -16,9 +17,25 @@ const settings = getSetting( 'cheque_data', {} );
 
 const EditPlaceHolder = () => <div>TODO: Edit preview soon...</div>;
 
-const Content = ( props ) => {
-	const { activePaymentMethod } = props;
+/**
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ */
 
+/**
+ * Cheque content component
+ *
+ * @param {RegisteredPaymentMethodProps|Object} props Incoming props
+ */
+const Content = ( { activePaymentMethod, eventRegistration } ) => {
+	// hook into payment processing event.
+	useEffect( () => {
```

Comment:
Can you help me understand what this does exactly?

Can we abstract/import it so gateways have less to worry about?

Reply:
Basically every gateway will have to at a _minimum_ register an observer on the `onPaymentProcessing` event.  This event fires when checkout has initiated payment processing as a part of the flow.  What gateways return from this callback when it's invoked determines whether the gateway was succesful, failed, or errored for it's processing. The possible shapes of the return value are:

### Success
Can return:
- `true`
- OR an object: `{ paymentMethodData, billingData }`

### Fail:
Return an object: `{ fail: { errorMessage, paymentMethodData } }`

### Error:
Return an object: `{ errorMessage, validationErrors }`

- errorMessage is used as just a general error message to display in the payment method step.
- validationErrors is an object where keys are field ids and values are the error message for that field (to help with displaying field validation errors for for the payment method).

Reply:
> Can we abstract/import it so gateways have less to worry about?

So one thing I was thinking we could do is have a function for registering really simple gateways that are similar to cheque in that they all the processing happens server side. So it'd have a fairly simple signature (gateway name, maybe icon, and text for the content window). Then this function would take care of the payment method registration (with `Config` etc) using components common to all these payment methods.  We could export this as a helper on the block-registry global.